BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1514,6 +1491,7 @@ protected override void ConnectImpl()
                 adev.AdvancedModeEnabled = true;
 
             _depthScale = GetDepthScale();
+            DecimationFilter.ResolutionChanged += UpdateImpl;
```

Comment:
I could see an issue with this mechanism:

Modifying the DecimationFilter will call `UpdateImpl` which means the image content returned by `Calc*` will be changed afterwards. But the image's `FrameNumber` will not have been updated (because that's done in `Camera.Update`)

Reply:
Alternatively we could call `Update()` instead.

Another idea would be to filter the current frame - if available - as soon as the filter is enabled. But for this to work properly we would need to keep a copy of the unfiltered frame to account for magnitude changes and deactivating the filter again. Also would mean having to call update if it hasn't been called before.
But all of this sounds much more complicated and error prone to me.